java/google/registry/rde/imports/XjcToDomainResourceConverter.java

```diff
@@ -293,16 +238,34 @@ private static HistoryEntry createHistoryEntry(XjcRdeDomain domain) {
   }
 
   /** Converts {@link XjcRdeDomainTransferDataType} to {@link TransferData}. */
-  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data) {
+  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data,
+      DateTime domainExpiration) {
     if (data == null) {
       return TransferData.EMPTY;
     }
+    // If the transfer is pending, calculate the number of years to add to the domain expiration
```

Comment:
This calculation will need to subsume an autorenew's extra year if the domain will still be within the autorenew grace period at the automatic transfer time.  However, there are existing issues with how our transfer flows implement this, so I will take on figuring out a consistent fix for both the EPP flows and here - this is just FYI.